cli/src/personhood_oracle/commands/fetch_reputation.rs

```diff
@@ -0,0 +1,117 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+		http://www.apache.org/licenses/LICENSE-2.0
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+use crate::{
+	command_utils::{get_accountid_from_str, get_chain_api, get_worker_api_direct},
+	Cli,
+};
+use codec::{Decode, Encode};
+use encointer_primitives::{
+	ceremonies::Reputation, communities::CommunityIdentifier, scheduler::CeremonyIndexType,
+};
+use itc_rpc_client::direct_client::DirectApi;
+use itp_node_api::api_client::ParentchainApi;
+use itp_rpc::{RpcRequest, RpcResponse, RpcReturnValue};
+
+use itp_types::DirectRequestStatus;
+use itp_utils::FromHexPrefixed;
+use log::error;
+
+use std::str::FromStr;
+use substrate_api_client::GetStorage;
+
+#[derive(Debug, Clone, Parser)]
+pub struct FetchReputationCmd {
+	pub account: String,
+	pub cid: String,
+	pub number_of_reputations: CeremonyIndexType,
+}
+
+impl FetchReputationCmd {
+	pub fn run(&self, cli: &Cli) {
+		if let Ok(reputations) = self.fetch_reputation_rpc(cli) {
+			let verified_reputations = reputations.iter().filter(|rep| rep.is_verified()).count();
+			println!("reputation for {} is: {:#?}", &self.account, reputations);
+			println!(
+				"verified reputatations number: {} out of:{}",
+				verified_reputations,
+				reputations.len()
+			);
+		}
+	}
+
+	pub fn fetch_reputation_rpc(&self, cli: &Cli) -> Result<Vec<Reputation>, String> {
```

Comment:
This is just a nitpick, but this method doesn't need self, and hence I would prefer if it is a standalone function to stay consistent with the design.

Reply:
self is used in `self.number_of_reputations.encode(),`, but it can be removed once it has been redesigned and the reputation fetching is not done by the PO but via a user supplied `ProofOfAttendance`. I will add a `TODO` here as well.

Reply:
Super, I can live with that!